core/src/mender-client.c

```diff
@@ -1080,11 +1080,27 @@ mender_client_update_work_function(void) {
     /* reset the currently used update module */
     mender_update_module = NULL;
 
+    {
+        cJSON *deployment_id_j = cJSON_GetObjectItem(mender_client_deployment_data, "id");
```

Comment:
I have just moved it around, so there's not really a difference from before - but nothing happens if you pass a nullpointer here, it just returns NULL, maybe I am missing your point

Reply:
> nothing happens if you pass a nullpointer here, it just returns NULL, maybe I am missing your point

oh, no problem then. I thought it would crash. :cool: